mlflow/transformers/model_io.py

```diff
@@ -110,17 +110,37 @@ def load_model_and_components_from_huggingface_hub(flavor_conf, accelerate_conf,
 def _load_component(flavor_conf, name, local_path=None):
     import transformers
 
-    cls = getattr(transformers, flavor_conf[FlavorKey.COMPONENT_TYPE.format(name)])
+    _COMPONENT_TO_AUTOCLASS_MAP = {
+        FlavorKey.TOKENIZER: transformers.AutoTokenizer,
+        FlavorKey.FEATURE_EXTRACTOR: transformers.AutoFeatureExtractor,
+        FlavorKey.PROCESSOR: transformers.AutoProcessor,
+        FlavorKey.IMAGE_PROCESSOR: transformers.AutoImageProcessor,
+    }
```

Comment:
ref: https://huggingface.co/docs/transformers/v4.38.2/en/model_doc/auto#auto-classes

looks like these are all the available Auto[Components]:
<img width="218" alt="Screenshot 2024-03-15 at 12 48 23 PM" src="https://github.com/mlflow/mlflow/assets/148037680/4856b91b-9a03-40c2-9f5c-a27e7250a727">

(AutoConfig is used by AutoModel, and is handled in the model load function, see line 155 below)